app/src/test/java/tech/relaycorp/gateway/domain/courier/ProcessCargoTest.kt

```diff
@@ -0,0 +1,72 @@
+package tech.relaycorp.gateway.domain.courier
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.Test
+import tech.relaycorp.gateway.data.disk.CargoStorage
+import tech.relaycorp.gateway.data.model.MessageAddress
+import tech.relaycorp.gateway.data.model.MessageId
+import tech.relaycorp.gateway.data.model.RecipientLocation
+import tech.relaycorp.gateway.domain.DeleteParcel
+import tech.relaycorp.gateway.domain.StoreParcel
+import tech.relaycorp.gateway.domain.StoreParcelAck
+import tech.relaycorp.gateway.test.factory.CargoFactory
+import tech.relaycorp.gateway.test.factory.ParcelCollectionAckFactory
+import tech.relaycorp.gateway.test.factory.ParcelFactory
+import tech.relaycorp.gateway.test.factory.StoredParcelFactory
+import tech.relaycorp.relaynet.messages.payloads.CargoMessage
+
+class ProcessCargoTest {
```

Comment:
I'd suggest to add tests for:

- Invalid/unsupported messages in cargoes.
- Malformed parcels.
- Well-formed but invalid parcels.